controllers/masteruserrecord/sync_test.go

```diff
@@ -235,6 +235,8 @@ func TestSyncMurStatusWithUserAccountStatusWhenUpdated(t *testing.T) {
 
 		// when
 		preSyncTime := metav1.Now()
+		// add a delay of a second to avoid flakiness
+		time.Sleep(1 * time.Second)
```

Comment:
@MatousJobanek This particular test would fail periodically. And looking at the test further on line 244/242 
`require.True(t, preSyncTime.Time.Before(sync.record.Status.ProvisionedTime.Time), "the timestamp just before syncing should be before the ProvisionedTime")`
Adding a delay should only be ensuring that the preSyncTime is before the provisionedTime, so i thought it should be okay to add.

Reply:
ok, but I would love to understand what has changed that it didn't fail before, but fails now.

Reply:
it isn't something obvious, so I'm still trying to decode what changed
But I can see that with my changes the provision time is truncated to 2 decimal places and that why the time.Before fails
master:
`presync time : 2024-08-12 16:02:04.93909 -0400 EDT m=+0.017812959
  provisioned Time 2024-08-12 16:02:04.939669 -0400 EDT m=+0.018391543`
PR:
`INFO	 presyncTime : 2024-08-12 16:37:18.01163 -0400 EDT m=+0.015583334
  INFO	 provisioned Time : 2024-08-12 16:37:19 -0400 EDT`

Reply:
its in [this](https://github.com/kubernetes-sigs/controller-runtime/commit/0bff6fc8af438727a2a9e5c8a2050fdad25952b4#diff-20ecedbf30721c01c33fb67d911da11c277e29990497a600d20cb0ec7215affdR882-R894) function that the timeStamp is truncated in updating

Reply:
I see, there was a change in the way it is being unmarshalled, so the meta time contains seconds as the smallest unit. Ok, in that case the time.sleep makes complete sense :+1: 
Thanks for the investigation and for the info :+1: